### PR TITLE
Narcissistic numbers printer implementation

### PR DESCRIPTION
**Pull request by krajasekhar-unify**
This is the solution for the assignment - Narcissistic numbers.
- Solution to printing narcissistic numbers for the range given from the input.
- Added unit test cases to test the narcissistic checker and printer.

### CODE REVIEW

File: src/rajasekhar/narcissistic/index.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+ * It is also sometimes known as an Armstrong number
+ * Example:
+ * 	153 = (1*1*1)+(5*5*5)+(3*3*3)
+ * 	1634 = (1*1*1*1)+(6*6*6*6)+(3*3*3*3)+(4*4*4*4)
+ * Problem Statement: Find all the Narcissistic Numbers between 10 to 10000
+ * @param {number} start - starting number of range
+ * @param {number} end - ending number of range
+ * @return {number[]} returns an array of numbers which are narcissistic
+ */
+export const narcissisticPrinter = (start = 10, end = 10000): Array<number> => {
```
Comment (nirajunify) [inline src/rajasekhar/narcissistic/index.ts]: Really liked the usage of default numbers

Comment (krajasekhar-unify) [reply src/rajasekhar/narcissistic/index.ts]: I see that type definition won't make sense when we are having a default assignment. My type checks for the arguments where auto removed in following the rule `@typescript-eslint/no-inferrable-types`.

**Review by nirajunify — CHANGES_REQUESTED** [review]
If you are using the directory pattern then index.ts should contain only the import and export of functions written in a separate file - narcissisticNumbers.ts
You should rename your test.ts to narcissisticNumber.<spec|test>.ts

**Comment by krajasekhar-unify** [comment]
I agree to rename my test file to `narcissisticNumber.test.js` but why shall I not keep my main logic (the function which I am responding to out of the folder) in `index.ts`? Why to make an extra file with keeping only `export` & `import` in the index file? Any reason behind this?
I agree if there are multiple functions need to be exported from a single folder where we keep each function in a different file and we need an index file to group them and export together. Shall we still do it if I am exporting one function for my folder?
By keeping the logic in `index.ts` at the scenario of one function to export, When I am navigating (Ctrl + p), I can directly land to my logic by typing folder name. Thoughts? 
Screenshot from 2020-09-12 12-40-57

**Comment by nirajunify** [comment]
> why shall I not keep my main logic (the function which I am responding to out of the folder) in `index.ts`? Why to make an extra file with keeping only `export` & `import` in the index file? Any reason behind this?

Yes. This is a best practice. The idea is that outsiders, developers who use our package, should not meddle with its internal structure, search for files inside our package folder. We export only what’s necessary in auth/index.js and keep the rest hidden from prying eyes.

> Shall we still do it if I am exporting one function for my folder?

You should be mindful about best practices even when you think your code is very little. It's about getiing into the habit

**Comment by zulqer-nain** [comment]
I think if there is more than one context and more than one  function of every context then we should maintain files of functions and maintain index file to import/export all the function .
In case Narcissistic problem .
best directory structure should be :
 
index ts - code_compete - Visual Studio Code 14-09-2020 10_01_44 (2)

**Comment by krajasekhar-unify** [comment]
Yes. It is good to move the tests to a different folder as it separates the test code from parent logic. But for the test file in `__tests__` folder, we don't the require the name as `index.test.js` since we are not importing it anywhere and the tests are auto picked. Also it will be bit difficult to navigate to it from `search/Ctrl+P` by typing filename. So I advice to still keep the test file name as `narcissistic.test.js`.

**Review by zulqer-nain — DISMISSED** [review]
Name of function "narcissisticPrinter" can be improved 
I am approving PR

**Comment by krajasekhar-unify** [comment]
@nirajunify mentioned there shall be index.ts in every folder whose job shall be just to do grouping of functions from other files in the folder. Any thoughts? @zulqer-nain

**Review by nirajunify — CHANGES_REQUESTED** [review]
Unit Test file reside beside the primary file with .test extension . Do not create __tests__ directory

**Review by nirajunify — APPROVED** [review]

**Pull request merged**